### ISSUE TITLE
Which asset extractor did you use?

### ISSUE THREAD

**Issue by shinich39** [mos9527/sssekai#14]
Some images failed to extract.
I used Asset Ripper and Asset Studio.
Asset Ripper is missing files more than Asset Studio.

OS: windows11.
App Version: 4.1.0
App region: jp
Unity Settings: 2022.3.21f1

I don't remember exactly but byte -1 is in the error message.

**Comment by mos9527** [comment]
I've mostly used [Razmoth's fork of AssetStudio](https://github.com/RazTools/Studio) besides [UnityPy](https://github.com/K0lb3/UnityPy)
With the latter I've also used it in [sssekai_blender_io](https://github.com/mos9527/sssekai_blender_io) that does involve image/`Texture2D` conversion

Technically (for Unity textures) the latest versions UnityPy should offer the best compatibility since it's using the de facto [ASTC texture codec implementation](https://github.com/K0lb3/astc-encoder-py/tree/main) whilst using [AssetStudio's Texture2D converter](https://github.com/K0lb3/texture2ddecoder) to handle other formats
The game (just like most, if not all mobile Unity games) uses ASTC for the majority of its texture encoding - I'd assume it's one of these textures that didn't work on AssetStudio/AssetRipper since their implementations of ASTC codec *probably* doesn't offer the same robustness of the offical one which could cause them to fail on some edge cases:
- https://github.com/Perfare/AssetStudio/blob/master/Texture2DDecoderNative/astc.cpp
- https://github.com/AssetRipper/TextureDecoder/blob/master/AssetRipper.TextureDecoder/Astc/AstcDecoder.cs

I've dug around a bit and found some related issues as well,which again suggests it's the issues with the said format.
- https://github.com/AssetRipper/AssetRipper/issues/1119
- https://github.com/AssetRipper/TextureDecoder/issues/59 (deleted for some reason)

TL;DR try [Raz's fork](https://github.com/RazTools/Studio) or dump the textures with [UnityPy](https://github.com/K0lb3/UnityPy?tab=readme-ov-file#texture2d). BTW I'd be interested in the files that don't work if you can post them here too^^

**Comment by shinich39** [comment]
Thanks for your answer and cool program!
I will test it in the next version.
